compiler/ast.rs

```diff
@@ -0,0 +1,633 @@
+// Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
+
+use crate::msg::MediaType;
+use crate::Result;
+
+use deno_core::ErrBox;
```

Comment:
Note to self, most of the contents of this file are used in `cli/` (mainly for compiler purposes but also for `deno doc`) as well as in `deno_lint`. It should be made into a utility crate that is shared between the other crates.